### PR TITLE
core: merge buffer by name

### PR DESCRIPTION
**Pull request by KryDos**
This is the PR for #1108

Unfortunately it covers the issue partially. I'm not really sure how it's possible to unmerge by name since buffer number is involved. It would be great if someone can help me to understand it. 

I also guess I need to update docs somewhere, but I'm not really sure where.

### CODE REVIEW

**Comment by sim642** [comment]
The first line of documentation to be changed would be [here on the bottom of the same file](https://github.com/KryDos/weechat/blob/bbbb81d614ea5ee815a93d3641b5f2ef3c052803/src/core/wee-command.c#L7094) where the `/buffer` command is hooked. There the `/help buffer` format and description are defined.

I only looked deeply in the code what the `/buffer unmerge` number argument does just now and it's not similar to `/buffer merge` at all. Namely, the argument for unmerging specifies the buffer number you want the buffer to have after it's unmerged, not which buffer to unmerge (that's always the current one). Also turns out `/help` doesn't really explain the number's meaning for this subcommand either. In this interpretation taking a name argument doesn't make sense anyway. Of course this behavior could be changed but that would break compatibility which isn't a good thing. This is probably the cause for not understanding the buffer number argument for unmerging.

File: src/core/wee-command.c [hunk 1]
```diff
@@ -792,27 +792,23 @@ COMMAND_CALLBACK(buffer)
...
+        ptr_buffer = gui_buffer_search_by_number_or_name (argv[2]);
+
+        if (!ptr_buffer)
         {
-            /* invalid number */
-            gui_chat_printf (NULL,
-                             _("%sError: incorrect buffer number"),
-                             gui_chat_prefix[GUI_CHAT_PREFIX_ERROR]);
-            return WEECHAT_RC_OK;
+          gui_chat_printf (NULL,
```
Comment (sim642) [inline src/core/wee-command.c]: Indentation of this block is inconsistent.

**Comment by KryDos** [comment]
hi everyone. Thanks for your feedback and sorry about messing up with the indentation. 
I've fixed indentation and updated the doc. 

As @sim642 mentioned it's not make much sense now to make unmerge command to work with buffer name. At least it's quite easy todo without breaking backward compatibility. 

So, let's keep unmerge command untouched?

**Comment by KryDos** [comment]
ping

**Comment by flashcode** [comment]
Thanks, merged with minor changes.

**Pull request closed without merging**